pkg/sources/git/git.go

```diff
@@ -417,10 +418,11 @@ func executeClone(ctx context.Context, params cloneParams) (*git.Repository, err
 	}
 
 	gitArgs := []string{
-		"clone",
-		cloneURL.String(),
+		"clone", cloneURL.String(),
 		params.clonePath,
+		"-c", "remote.origin.fetch=+refs/*:refs/remotes/origin/*",
```

Comment:
This is already gated by the flag below; if that's not working right now we should fix it :/